### PR TITLE
Add session data guard (GitHub Issue #672).

### PR DESCRIPTION
**Pull request by E-B-Smith**
Added a guard for session data. Sometimes the session data is not stringified (an error on the backend) but this guards against a crash in the SDK.

### CODE REVIEW

**Comment by BranchMacMini** [comment]
Result of [Integration 1](https://stlt.herokuapp.com/v1/xcs_deeplink/qabot.stage.branch.io/0a5c72c332d8d1933d4f496573d460e4/0a5c72c332d8d1933d4f496573d467c2)
---
*Duration*: 9 minutes and 17 seconds
*Result*: **Build failed 1 test** out of 187

File: Branch-SDK/Branch-SDK/Requests/BranchOpenRequest.m [hunk 1]
```diff
@@ -109,6 +109,29 @@ - (void)processResponse:(BNCServerResponse *)response error:(NSError *)error {
     [BNCSystemObserver setUpdateState];
 
     NSString *sessionData = data[BRANCH_RESPONSE_KEY_SESSION_DATA];
+    if (sessionData == nil || [sessionData isKindOfClass:[NSString class]]) {
+    } else
+    if (([sessionData isKindOfClass:[NSDictionary class]] ||
+         [sessionData isKindOfClass:[NSArray class]]) &&
+            [NSJSONSerialization isValidJSONObject:sessionData]) {
```
Comment (ahmednawar) [inline Branch-SDK/Branch-SDK/Requests/BranchOpenRequest.m]: Should this code reside in BNCEncodingUtils similar to https://github.com/BranchMetrics/ios-branch-deep-linking/pull/674/files#diff-4a8bffd322f48b39ef8419977f64ae8bR146?

**Comment by BranchMacMini** [comment]
Result of [Integration 3](https://stlt.herokuapp.com/v1/xcs_deeplink/qabot.stage.branch.io/0a5c72c332d8d1933d4f496573d460e4/b53de82f295e7dcac1c9985afe5ee852)
---
*Duration*: 7 minutes and 46 seconds
*Result*: **Perfect build!** All 187 tests passed. :+1:
*Test Coverage*: 58%

**Comment by E-B-Smith** [comment]
@ahmednawar I updated the JSON convertion to the standard methods. Can you thumb this?

**Comment by ahmednawar** [comment]
👍

**Pull request merged**